db/data_migrate/20190325121200_fix_us_formatted_dates.rb

```diff
@@ -0,0 +1,72 @@
+require 'progress_bar'
+require 'string_utils'
+
+puts 'Fixing submission entries...'
+
+affected_submission_ids = SubmissionEntry
+                          .validated
+                          .distinct
+                          .select(:submission_id)
+                          .joins(:submission)
+                          .where(
+                            "(LENGTH(data->>'Customer Invoice Date') IN (6,7,8) " \
+                            "OR LENGTH(data->>'Customer Invoice/Credit Note Date') IN (6,7,8))"
+                          )
+                          .where.not(submissions: { aasm_state: 'validation_failed' })
+                          .pluck(:submission_id)
+
+bar = ProgressBar.new(affected_submission_ids.count)
+
+affected_submission_ids.each do |submission_id|
+  submission_entries = SubmissionEntry
+                       .where(submission_id: submission_id)
+
+  target_fields = submission_entries.flat_map do |entry|
+    entry.data.keys.select { |k| k.match?(/Date/) }
+  end.compact.uniq
+
+  dates_in_submission = submission_entries.flat_map do |entry|
+    entry.data.select { |k, v| k.match?(/Date/) && v.size > 5 && v.size < 9 }.values
+  end.compact.uniq
+
+  # Detect date format
+  first_component_maybe_day = dates_in_submission.any? { |date| date.split('/')[0].to_i > 12 }
+  second_component_maybe_day = dates_in_submission.any? { |date| date.split('/')[1].to_i > 12 }
+
+  input_format = if first_component_maybe_day && second_component_maybe_day
```

Comment:
I am probably missing something: what happens if neither the first nor the second component are >12? There's no way to distinguish '10/11/18' (US) from '11/10/18' (UK), so shouldn't it say that it's `:unknown`?

Reply:
:+1:

I think when testing this never actually happened (which suprised me!), but will add it back for completeness.